### PR TITLE
Add x16r and x16rv2 algorithms.

### PR DESCRIPTION
**Pull request by Stackout**

### CODE REVIEW

**Comment by Stackout** [comment]
Can we get this merged before Oct 1st?

**Comment by leshacat** [comment]
Can we get this pulled ASAP please, @zone117x @foxer666 

And please keep eyes open for PR from @Stackout thanks

File: sha3/sha2.c [hunk 1]
```diff
@@ -0,0 +1,690 @@
+/* $Id: sha2.c 227 2010-06-16 17:28:38Z tp $ */
```
Comment (zone117x) [inline sha3/sha2.c]: This appears to be a duplicate of `sha3/sph_sha2.c`?

File: sph_sha2.h [hunk 2]
```diff
@@ -0,0 +1,370 @@
+/* $Id: sph_sha2.h 216 2010-06-08 09:46:57Z tp $ */
```
Comment (zone117x) [inline sph_sha2.h]: Is this a duplicate of `sha3/sph_sha2.h` ?

**Comment by Stackout** [comment]
This is ready for review. I needed some time to test it again after my change.

**Pull request merged**